plugins/ci/skills/archive-payload-result/extract_session_data.py

```
@@ -0,0 +1,441 @@
+            "If payload-tag is omitted, attempts to detect it from session content."
+        )
+        sys.exit(1)
+
+    tarball = sys.argv[1]
+    output_dir = sys.argv[2]
+    payload_tag = sys.argv[3] if len(sys.argv) > 3 else None
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        print(f"Extracting {tarball}...")
+        with tarfile.open(tarball) as tf:
+            tf.extractall(tmpdir)
```

Ruff (0.15.12)

[error] Uses of `tarfile.extractall()`

(S202)